marketplaceItems/audioFocus/appResources/appData/audioFocus_app.js

```diff
@@ -0,0 +1,459 @@
+        if (handController === Controller.Standard.LeftHand) {
+            offset = {
+                x: -GRAB_POINT_SPHERE_OFFSET.x,
+                y: GRAB_POINT_SPHERE_OFFSET.y,
+                z: GRAB_POINT_SPHERE_OFFSET.z
+            };
+        }
+        if (ignoreSensorToWorldScale) {
+            return offset;
+        } else {
+            return Vec3.multiply(MyAvatar.sensorToWorldScale, offset);
+        }
+    }
+
+    // controllerWorldLocation is where the controller would be, in-world, with an added offset
+    function getControllerWorldLocation(handController, doOffset) {
+        var orientation;
+        var position;
+        var valid = false;
+
+        if (handController >= 0) {
+            var pose = Controller.getPoseValue(handController);
+            valid = pose.valid;
+            var controllerJointIndex;
+            if (pose.valid) {
+                if (handController === Controller.Standard.RightHand) {
+                    controllerJointIndex = MyAvatar.getJointIndex("_CAMERA_RELATIVE_CONTROLLER_RIGHTHAND");
+                } else {
+                    controllerJointIndex = MyAvatar.getJointIndex("_CAMERA_RELATIVE_CONTROLLER_LEFTHAND");
+                }
+                orientation = Quat.multiply(MyAvatar.orientation, MyAvatar.getAbsoluteJointRotationInObjectFrame(controllerJointIndex));
+                position = Vec3.sum(MyAvatar.position, Vec3.multiplyQbyV(MyAvatar.orientation, MyAvatar.getAbsoluteJointTranslationInObjectFrame(controllerJointIndex)));
+
+                // add to the real position so the grab-point is out in front of the hand, a bit
+                if (doOffset) {
+                    var offset = getGrabPointSphereOffset(handController);
+                    position = Vec3.sum(position, Vec3.multiplyQbyV(orientation, offset));
+                }
+
+            } else if (!HMD.isHandControllerAvailable()) {
+                // NOTE: keep this offset in sync with scripts/system/controllers/handControllerPointer.js:493
+                var VERTICAL_HEAD_LASER_OFFSET = 0.1 * MyAvatar.sensorToWorldScale;
+                position = Vec3.sum(Camera.position, Vec3.multiplyQbyV(Camera.orientation, { x: 0, y: VERTICAL_HEAD_LASER_OFFSET, z: 0 }));
+                orientation = Quat.multiply(Camera.orientation, Quat.angleAxis(-90, { x: 1, y: 0, z: 0 }));
+                valid = true;
+            }
+        }
+
+        return {
+            position: position,
+            translation: position,
+            orientation: orientation,
+            rotation: orientation,
+            valid: valid
+        };
+    }
+
+    // #endregion
+    // *************************************
+    // END UTILITY FUNCTIONS
+    // *************************************
+
+
+    // *************************************
+    // START INIT
+    // *************************************
+    // #region Init
+
+    var isEnabled = true;
+    var soloAvatars = {};
```

Comment:
please move this closer to where it is first used (or where it's first relevant). perhaps right above `function soloAvatar(avatarUUID)`?